packages/taiko-client/driver/preconf_blocks/server.go

```diff
@@ -627,19 +658,37 @@ func (s *PreconfBlockAPIServer) ImportMissingAncientsFromCache(
 
 				log.Info(
 					"Publish preconfirmation block request",
-					"blockID", uint64(currentPayload.Payload.BlockNumber-1),
+					"blockID", parentNum,
 					"hash", currentPayload.Payload.ParentHash.Hex(),
 				)
 
-				if err := s.p2pNode.GossipOut().PublishL2Request(ctx, currentPayload.Payload.ParentHash); err != nil {
-					log.Warn(
-						"Failed to publish preconfirmation block request",
-						"blockID", uint64(currentPayload.Payload.BlockNumber-1),
-						"hash", currentPayload.Payload.BlockHash.Hex(),
-						"error", err,
-					)
+				publishRequest := func() {
+					if err := s.p2pNode.GossipOut().PublishL2Request(ctx, currentPayload.Payload.ParentHash); err != nil {
+						log.Warn(
+							"Failed to publish preconfirmation block request",
+							"blockID", parentNum,
+							"hash", currentPayload.Payload.BlockHash.Hex(),
+							"error", err,
+						)
+					} else {
+						s.blockRequestsCache.Add(currentPayload.Payload.ParentHash, struct{}{})
+					}
+				}
+				if s.latestSeenProposal != nil {
+					s.mutex.Lock()
+					tip := s.latestSeenProposal.Pacaya().GetLastBlockID()
+					s.mutex.Unlock()
+					if tip >= requestSyncMargin && parentNum <= tip-requestSyncMargin {
```

Comment:
the main change - here we dont publish requests for old blocks, flooding the network.